HouseRules_Core/LifecycleDirector.cs

```diff
@@ -61,6 +72,27 @@ private static void GameStartup_InitializeGame_Postfix(GameStartup __instance)
             _gameContext = gameContext;
         }
 
+        private static void CreatingGameState_TryCreateRoom_Prefix()
+        {
+            var createGameMode = Traverse.Create(_gameContext.gameStateMachine)
+                .Field<CreateGameMode>("createGameMode").Value;
+            if (createGameMode != CreateGameMode.Private)
+            {
+                return;
+            }
+
+            var gameStateTraverse = Traverse.Create(_gameContext.gameStateMachine).Field("creatingGameState");
+            if (!gameStateTraverse.FieldExists())
+            {
+                CoreMod.Logger.Error("Failed to find required \"creatingGameState\" field.");
+                return;
+            }
+
+            var gameState = gameStateTraverse.GetValue();
+            var roomOptions = Traverse.Create(gameState).Field<RoomOptions>("roomOptions").Value;
+            AddModdedRoomProperties(roomOptions);
```

Comment:
Great catch!  Indeed, I forgot to check to make sure HR is even active for these games.  I'll get that fixed so that the label is only applied when HR is in use.

---

General info regarding the room property:

Demeo's multiplayer network has a notion of a "Lobby" and a "Room".

A room corresponds to a Demeo game, while a lobby corresponds to the pool of rooms available to the user (and the pool a user's created room is added to).

The connection string that RG uses, which includes the game version, is used to connect to a network **_lobby_**.  So users with the same Demeo version all connect to the same lobby and thus have access to the same set of rooms.

Currently, RG labels their **_rooms_** with only three custom properties:
- Game type
- Current floor
- Game state (playing, shopping, postgame, etc.)

This is in addition to some default network platform properties for rooms (e.g., player count, max players, name, isOpen, etc.).

When a player's clients retrieves a list of all rooms from the network, it does so at a lobby-level (a client is given all public/private rooms, including ones they're banned from).  The game itself will then filter these rooms given some set of criteria (public only, game type that matches what the user selected).  It's these rooms that are then selected from when the user "quickjoins".  This is the same pool of rooms that RoomFinder uses for its listing.

This PR is proposing an additional (custom) room-level label.  One which, when set, won't affect whether or not the room is made available to users based on game version (since it's the lobby that determines that).  It also won't affect regular client filtering for quickjoins (since this label is custom and thus ignored by Demeo ... although it wouldn't matter since our games are always private and thus always filtered client-side by Demeo).

RoomFinder, (or any other mod) could look at the list of pre-filtered rooms (e.g., both public and private) and cherry-pick ones that are labelled as "modded" even if they're private.  It could then add these entries to the regular post-filtered (e.g., public) rooms that it lists.